### PR TITLE
add patient reactivation

### LINKED ISSUE #397: Reactivate discharged patients
### Describe the feature

Reactivate discharged patients by removing the soft delete flag

### Why is this feature needed?

Discharged patients still exist, but can't be used anymore, thus a reactivation would required

### Additional context
```proto
//
// ReactivatePatient
//

message ReactivatePatientRequest {
  string id = 1; // @gotags: validate:"uuid4"
}

message ReactivatePatientResponse {}
```

Note: You will need to use the `db.Unscoped().<query>` to be able to interact with soft deleted patients

### PR DESCRIPTION
**Pull request by DasProffi**
## Which issues does this pull request close?
closes #397

### CODE REVIEW

File: services/task-svc/internal/repositories/patient_repository.go [hunk 1]
```diff
@@ -130,3 +130,13 @@ func (r *PatientRepository) DeletePatient(patientID uuid.UUID) (*models.Patient,
 	}
 	return patient, nil
 }
+
+func (r *PatientRepository) ReactivatePatient(patientID uuid.UUID) (*models.Patient, error) {
+	patient := &models.Patient{ID: patientID}
+	query := r.db.Unscoped().Model(patient).Update("is_discharged", 0)
```
Comment (FoseFx) [inline services/task-svc/internal/repositories/patient_repository.go:136]: is `Unscoped` needed here? from what I can gather, it only relates to gorm's soft-delete mechanic

File: proto/services/task_svc/v1/patient_svc.proto [hunk 2]
```diff
@@ -227,6 +231,16 @@ message GetPatientListResponse {
 }
 
 
+//
+// ReactivatePatient
+//
+
+message ReactivatePatientRequest {
+  string id = 1; // @gotags: validate:"uuid4"
```
Comment (FoseFx) [inline proto/services/task_svc/v1/patient_svc.proto:239]:
```suggestion
  string patient_id = 1; // @gotags: validate:"uuid4"
```

File: proto/services/task_svc/v1/patient_svc.proto [hunk 3]
```diff
@@ -244,4 +258,5 @@ service PatientService {
   rpc GetPatientDetails(GetPatientDetailsRequest) returns (GetPatientDetailsResponse) {}
   rpc GetPatientList(GetPatientListRequest) returns (GetPatientListResponse) {}
   rpc DeletePatient(DeletePatientRequest) returns (DeletePatientResponse) {}
+  rpc ReactivatePatient(ReactivatePatientRequest) returns (ReactivatePatientResponse) {}
```
Comment (FoseFx) [inline proto/services/task_svc/v1/patient_svc.proto:261]: not the biggest fan of the name, `AdmitPatient` maybe?

**Review by FoseFx — APPROVED** [review]

File: services/task-svc/internal/repositories/patient_repository.go (hunk 1 above)
Comment (DasProffi) [reply services/task-svc/internal/repositories/patient_repository.go:136]: Yes, and as we look for the soft deleted patients to reactivate, we need the `Unscope`

File: proto/services/task_svc/v1/patient_svc.proto (hunk 3 above)
Comment (DasProffi) [reply proto/services/task_svc/v1/patient_svc.proto:261]: Me neither, maybe we should ask @aegis301

Comment (DasProffi) [reply proto/services/task_svc/v1/patient_svc.proto:261]: Maybe Readmittance?

**Pull request merged**